src/gpu/gpu_impl_list.hpp

```diff
@@ -77,7 +77,10 @@ namespace gpu {
 // NOTE: Support for the standalone GENERIC vendor has not been added yet.
 #if defined(DNNL_WITH_SYCL) \
         && ((DNNL_GPU_VENDOR == DNNL_VENDOR_GENERIC) \
-                || (DNNL_GPU_VENDOR == DNNL_VENDOR_NVIDIA))
+                || (DNNL_GPU_VENDOR == DNNL_VENDOR_INTEL) \
+                || (DNNL_GPU_VENDOR == DNNL_VENDOR_NVIDIA) \
+                || (DNNL_GPU_VENDOR == DNNL_VENDOR_AMD \
+                        && DNNL_HIP_DISABLE_SYCL_KERNELS != 1))
```

Comment:
I think it would be sufficient to use just `DNNL_HIP_DISABLE_SYCL_KERNELS` without checking for the vendor.

Reply:
I added a check in cmake to ensure both `DNNL_SYCL_HIP` is true and the architecture is specified for this to be set to 1 and removed the extra check for `VENDOR_AMD` here.